packages/api/endpoints/granules.js

```diff
@@ -307,39 +315,67 @@ const associateExecution = async (req, res) => {
   }
 
   const {
-    executionModel = new Execution(),
-    granuleModel = new Granule(),
+    executionPgModel = new ExecutionPgModel(),
+    granulePgModel = new GranulePgModel(),
+    collectionPgModel = new CollectionPgModel(),
     knex = await getKnexClient(),
     esClient = await Search.es(),
   } = req.testContext || {};
 
-  let granule;
-  let execution;
+  let pgGranule;
+  let pgExecution;
+  let pgCollection;
   try {
-    granule = await granuleModel.get({ granuleId });
-    execution = await executionModel.get({ arn: executionArn });
+    pgCollection = await collectionPgModel.get(
+      knex, deconstructCollectionId(collectionId)
+    );
+    pgGranule = await granulePgModel.get(knex, {
+      granule_id: granuleId,
+      collection_cumulus_id: pgCollection.cumulus_id,
+    });
+    pgExecution = await executionPgModel.get(knex, {
+      arn: executionArn,
+    });
   } catch (error) {
     if (error instanceof RecordDoesNotExist) {
-      if (granule === undefined) {
-        return res.boom.notFound(`No granule found to associate execution with for granuleId ${granuleId}`);
+      if (pgCollection === undefined) {
+        return res.boom.notFound(`No collection found to associate execution with for collectionId ${collectionId}`);
+      }
+      if (pgGranule === undefined) {
+        return res.boom.notFound(`No granule found to associate execution with for granuleId ${granuleId} and collectionId: ${collectionId}`);
+      }
+      if (pgExecution === undefined) {
+        return res.boom.notFound(`No execution found to associate granule with for executionArn ${executionArn}`);
       }
       return res.boom.notFound(`Execution ${executionArn} not found`);
     }
     return res.boom.badRequest(errorify(error));
   }
 
-  if (granule.collectionId !== collectionId) {
-    return res.boom.notFound(`No granule found to associate execution with for granuleId ${granuleId} collectionId ${collectionId}`);
-  }
-
-  const updatedGranule = {
-    ...granule,
-    execution: execution.execution,
-    updatedAt: Date.now(),
+  // Update both granule objects with new execution/updatedAt time
+  const updatedPgGranule = {
+    ...pgGranule,
+    updated_at: new Date(),
+  };
```

Comment:
What is actually associating with the PG granule with the execution specified in the request? I don't think anything is with this refactor. If I'm correct, then we need a unit test to cover that.

Reply:
It's a bit snarly, but the execution is associated via `writeGranuleRecordAndPublishSns`, by virtue of `executionCumulusId: pgExecution.cumulus_id` being passed down the call chain. 

`writeGranuleRecordAndPublishSns`->`_writeGranuleRecords`->`_writePostgresGranuleViaTransaction-`>`upsertGranuleWithExecutionJoinRecord`

It's functionally what was being done prior to the refactor. 

We also have a unit covering it, in the form of https://github.com/nasa/cumulus/blob/jk%2FCUMULUS-2768/packages/api/tests/endpoints/test-granules.js#L2521-L2522

That test validates that a call to the associate endpoint writes to dynamo and postgres.